wraps/python/src/cstypes/__init__.py

```diff
@@ -0,0 +1,6 @@
+import clr
+clr.AddReference(r'dll/Skender.Stock.Indicators')
```

Comment:
When it comes time to run the build, we'll do the C# build first, then somehow use that DLL when we package this.  I'd like to avoid manually copying the DLL to this location, if at all possible.

Reply:
Yes, this is the thing I want to discuss.
I hope so, but still don't know where is the proper location for the dll. I place the dll in `.../wrap/python/dll` temporarily to develop first. 

Also, the path string for `AddReference()` have to be change by the directory where I run the code.  but I still cannot figure out where to be set. Because it is module and it should be imported to any other directory where the users want.

Could you do some research for this? I couldn't find the solution until now.

Reply:
Are you able to use the .NET 5 version of the DLL?  If so, maybe we try `../../indicators/bin/Release/net5.0/` as the path, which will require a `dotnet build -c Release` when we're developing to produce it in its native location.  I'm hoping when we do the packaging, that relative path won't be an issue.

Reply:
I set `../../../indicators/bin/Release/net461/Skender.Stock.Indicators` as the path after doing `dotnet build -c Release`.
There is one problem. When I import the dll in `net5.0`, `netcoreapp3.1` and `netstandard2.1`, it shows the error, "ModuleNotFoundError: No module named 'Skender'", maybe python cannot recognize the module.
But when I import the dll in `net461` and `netstandard2.0`, it works successfully.
Do you know the reason?

Reply:
I suspect this is a Python.NET compatibility issue where .NET Standard 2.0 is the best they can support.  Here's some discussion on the topic:
`https://github.com/pythonnet/pythonnet/issues/857`

This may be the solutions?
https://pypi.org/project/clr-loader

Reply:
> but we'll have to figure out how to automatically update this DLL when the C# project builds;

I suddenly got a question about it. Did you mean auto-updating dll by network? The dll should be hard-copied at the end, I mean when we distribute this wrapper on PyPi.

Also understood that we should use NET Standard 2.0 at least now because of compatibility of Python .Net. But I couldn't find out what `clr-loader` is used for, for a few days. There is not enough description, just can guess on `https://github.com/pythonnet/pythonnet/issues/857`.

Anyway, there is another way we can set the path for dll on production code.
=> [2.6. Installing Package Data](https://docs.python.org/3/distutils/setupscript.html#installing-additional-files)

Examples:
`https://stackoverflow.com/questions/57191483/include-a-dll-in-python-module-pypi`
`https://stackoverflow.com/questions/33216691/add-a-dll-so-to-a-python-built-distribution`

Reply:
1. Version targeting: Ok, let's just use the .NET Standard 2.0 for now.  We can always revisit later when Python.NET upgrades itself to use newer .NET Core or .NET 5
2. DLL Loading: I was not thinking of the actual programatic loading, but glad you're thinking of that.  Mostly I wanted to make sure we're using the final version stamped DLL and not the one we might copy locally during development.  It's a chicken vs egg problem.  The final version stamped DLL is only produced during the build, so you'd never be able to copy it yourself beforehand.  I think what we can do is basically go back to your original dll folder, add the locally produced DLL manually, but then replace it with the stamped DLL during the final build (simple file copy command).

Reply:
When I place the dll in `.../wraps/python/lib`, example is like the below:
```python
# cstypes/__init__.py
import os
import sys
import clr

dir = os.path.dirname(sys.modules["cstypes"].__file__)
path = os.path.join(dir, "../../lib/Skender.Stock.Indicators.dll")
clr.AddReference(path)

```

Reply:
Now, I get it!! I was totally misunderstanding your words. OMG.. I was thinking that you want to update automatically dll in distribution..OK, Now, I know..

By the way, the below solves the problem about relative path, I mentioned above.

When I place the dll in `.../wraps/python/lib`, example is like the below:
```python
# cstypes/__init__.py
import os
import sys
import clr

dir = os.path.dirname(sys.modules["cstypes"].__file__)
path = os.path.join(dir, "../../lib/Skender.Stock.Indicators.dll")
clr.AddReference(path)

```